assets/new-user-experience/modules/advanced-settings/index.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * External dependencies.
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { ToggleControl } from '@wordpress/components';
+import parse from 'html-react-parser';
+
+/**
+ * Internal dependencies.
+ */
+import {
+	Section,
+	SectionTitle,
+	SectionDescription,
+	InputWrapper,
+	SquareCheckboxControl,
+} from '../../components';
+import { DebugMode } from '../../modules';
+import { useSquareSettings } from '../../settings/hooks';
+import { useSteps } from '../../onboarding/hooks';
+
+export const AdvancedSettings = () => {
+	const {
+		settings,
+		squareSettingsLoaded,
+		setSquareSettingData,
+	} = useSquareSettings();
+
+	const {
+		stepData: {
+			step,
+		}
+	} = useSteps();
+
+	const {
+		enable_customer_decline_messages,
+		debug_logging_enabled = 'no',
+	} = settings;
+
+	if ( ! ( squareSettingsLoaded ) ) {
+		return null;
+	}
+
+	return (
+		<>
+			<Section>
+				<SectionTitle title={ __( 'Advanced Settings', 'woocommerce-square' ) } />
+				<SectionDescription>
+					{ __( 'Adjust these options to provide your customers with additional clarity and troubleshoot any issues more effectively.', 'woocommerce-square' ) }
+					{ <br /> }
+					{ 'advanced-settings' === step && parse(
+						sprintf(
+							__( '%1sClick here%2s to further refine your settings in the traditional view.', 'woocommerce-square' ),
+							'<a href="/wp-admin/admin.php?page=wc-settings&tab=square">',
+							'</a>',
+						)
+					) }
+
+				</SectionDescription>
+
+				<div className='woo-square-wizard__fields'>
+					<InputWrapper
+						label={ __( 'Detailed Decline Messages', 'woocommerce-square' ) }
+					>
+						<SquareCheckboxControl
+							checked={ 'yes' === enable_customer_decline_messages }
+							onChange={ ( enabled ) => setSquareSettingData( { enable_customer_decline_messages: enabled ? 'yes' : 'no' } ) }
+							label={
+								__( 'Check to enable detailed decline messages to the customer during checkout when possible, rather than a generic decline message.', 'woocommerce-square' )
```

Comment:
Just my opinion but seems pretty obvious that they need to check this option to turn it on so not sure we need that in a label/description. I'd do something like:
```suggestion
								__( 'Show detailed decline messages to the customer during checkout rather than a generic decline message.', 'woocommerce-square' )
```